### PR TITLE
Revamp 3rd party app launcher to open up to more apps.

### PR DESCRIPTION
**Pull request by lytefast**
- dropbox and external apps are now selectable

### CODE REVIEW

**Comment by lytefast** [comment]
Should fix https://app.bugsnag.com/discord/discord-android/errors/58ffbca3e8cc1b00181d2f0e?filters%5Bevent.since%5D%5B%5D=30d&filters%5Berror.status%5D%5B%5D=open&filters%5Bversion.introduced_in%5D%5B%5D%5Bvalue%5D=3.7.3&filters%5Bversion.introduced_in%5D%5B%5D%5Btype%5D=eq&filters%5Bversion.introduced_in%5D%5B%5D%5Bvalue%5D=3.7.5&filters%5Bversion.introduced_in%5D%5B%5D%5Btype%5D=eq

File: flexinput/src/main/java/com/lytefast/flexinput/fragment/AddContentDialogFragment.java [hunk 1]
```diff
@@ -239,32 +239,59 @@ public void launchFileChooser() {
...
     startActivityForResult(chooserIntent, REQUEST_FILES);
   }
 
+  protected List<Intent> getAllIntents() {
+    final String[] mimetypes = {"text/*", "image/*", "video/*"};
+    List<ResolveInfo> resolveInfos = getContext().getPackageManager()
+        .queryIntentActivities(
+            new Intent(Intent.ACTION_GET_CONTENT)
+                .setType("application/*")
+                .putExtra(Intent.EXTRA_MIME_TYPES, mimetypes)
```
Comment (mrkcsc) [inline flexinput/src/main/java/com/lytefast/flexinput/fragment/AddContentDialogFragment.java:257]: I've read that using `EXTRA_MIME_TYPES` will not work on Android < 19 API, has this been tested?

https://issuetracker.google.com/issues/36987231

Comment (lytefast) [reply flexinput/src/main/java/com/lytefast/flexinput/fragment/AddContentDialogFragment.java:257]: yeah, it'll just be ignored. I tested with that ancient LG phone.

Comment (mrkcsc) [reply flexinput/src/main/java/com/lytefast/flexinput/fragment/AddContentDialogFragment.java:257]: Nice in that case it's win win.

**Review by mrkcsc — APPROVED** [review]

**Pull request merged**